### DISCUSSION TITLE
Timeline around support for descendent and sibling selectors

### DISCUSSION THREAD

**Discussion by keeganstreet** [facebook/stylex#465] [Q&A]
Hi there,

Comments like the following indicate the StyleX will at some point support descendent and sibling selectors:

> The limitation here is that we don't yet support descendent and sibling selectors. We're working on a proposal to share to add this feature soon.
> https://github.com/facebook/stylex/discussions/116#discussioncomment-7809774 

Are you able to share any information around when support for these will be added, or whether this is still on the roadmap?

More recent comments like the following make us wonder whether this feature is still on the roadmap:

> If you need selectors, do not use StyleX.
> https://github.com/facebook/stylex/issues/373#issuecomment-1936988070

Thanks

**Comment by nmn** [comment]
Yes, descendent and sibling selectors are still on the roadmap, but I want to make it clear that we will not be supporting arbitrary selectors. The functionality we're targeting is the same as `group` and `peer` from Tailwind. Other than the fact that we will allow multiple layers of groups/peers, we will not supporting any other functionality.

We will not be supporting "styling at a distance" and so you will never be able to apply styles on an element by "selector". Instead we're working on the ability to add a style conditionally when a parent is hovered, for example.

An RFC for this is overdue, so I will write it up within a day or two.
